schematic/utils/general.py

```diff
@@ -313,3 +314,68 @@ def normalize_path(path: str, parent_folder: str) -> str:
     if not os.path.isabs(path):
         path = os.path.join(parent_folder, path)
     return os.path.normpath(path)
+
+
+def create_like_statement(synapse_path: str) -> str:
+    """
+    Creates a sql like statement for a Synapse table query
+    See:
+      https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/web/controller/TableExamples.html
+
+    The statement is used to find all files in the folder, should be something like:
+      path like '<synapse_path>/%'
+
+    Certain special characters can also be used in like statements and these need to escaped with
+      a special character of the users choice. This function will use the '|' character like the
+      documentation shows. These need to have the "escape '|'" string added at the end, and
+      it will look like:
+
+      path like '<synapse_path>/%' escape '|'
+
+
+    Args:
+        synapse_path (str): The input synapse path to be made into a like statement
+
+    Raises:
+        ValueError: If the input path contains a '|' character
+
+    Returns:
+        str: A SQL like statement
+    """
+    if "|" in synapse_path:
```

Comment:
I could roughly follow the logic here, but can we add some comments in this section to make the logic clearer?